broker-daemon/block-order-worker/index.js

```diff
@@ -466,6 +472,8 @@ class BlockOrderWorker extends EventEmitter {
       throw new Error(`No engine available for ${counterSymbol}`)
     }
 
+    await this.checkFundsAreSufficient(blockOrder)
```

Comment:
you'll need to do this for each fill, since even here the block order doesn't have a `counterAmount`